### PR TITLE
add python and scala doc

### PR DESCRIPTION
**Pull request by yiheng**
## What changes were proposed in this pull request?
add python and scala doc

## How was this patch tested?
unit test

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/BatchNormalization.scala [hunk 1]
```diff
@@ -26,6 +26,26 @@ import com.intel.analytics.bigdl.utils.RandomGenerator._
...
+ * by Sergey Ioffe, Christian Szegedy https://arxiv.org/abs/1502.03167
+ *
+ * This implementation is useful for inputs NOT coming from convolution layers.
+ * For convolution layers, use nn.SpatialBatchNormalization.
+ *
+ * The operation implemented is:
+ *     y =     ( x - mean(x) )
+ *          -------------------- * gamma + beta
+ *             standard-deviation(x)
+ * where gamma and beta are learnable parameters.The learning of gamma and beta is optional.
```
Comment (SeaOfOcean) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/BatchNormalization.scala]: move y= from line 38 to line 39

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/dataset/image/BGRImgRdmCropper.scala [hunk 2]
```diff
@@ -25,6 +25,12 @@ object BGRImgRdmCropper {
     new BGRImgRdmCropper(cropHeight, cropWidth, padding)
 }
 
+/**
+ * Random crop a specified area from the Image. The result is also a image
+ * @param cropHeight crop area height
```
Comment (SeaOfOcean) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/dataset/image/BGRImgRdmCropper.scala]: an image

File: pyspark/dl/nn/layer.py [hunk 3]
```diff
@@ -586,14 +596,31 @@ def __init__(self,
...
+    by Sergey Ioffe, Christian Szegedy https://arxiv.org/abs/1502.03167
+
+    This implementation is useful for inputs NOT coming from convolution layers. For convolution
+    layers, use nn.SpatialBatchNormalization.
+
+    The operation implemented is:
+         y = ( x - mean(x) )
+             -------------------- * gamma + beta
+             standard-deviation(x)
+    where gamma and beta are learnable parameters.The learning of gamma and beta is optional.
```
Comment (SeaOfOcean) [inline pyspark/dl/nn/layer.py]: move y from 611 to 612

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/BatchNormalization.scala (hunk 1 above)
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/BatchNormalization.scala]: done

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/dataset/image/BGRImgRdmCropper.scala (hunk 2 above)
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/dataset/image/BGRImgRdmCropper.scala]: done

File: pyspark/dl/nn/layer.py (hunk 3 above)
Comment (yiheng) [inline pyspark/dl/nn/layer.py]: done

**Pull request merged**